services/codecov/codecov.service.js

```diff
@@ -20,19 +21,34 @@ const schema = Joi.object({
 
 const queryParamSchema = Joi.object({
   token: Joi.string(),
+  // https://docs.codecov.io/docs/flags
+  // Flags must be lowercase, alphanumeric, and not exceed 45 characters
+  flag: Joi.string().regex(/^[a-z0-9_]{1,45}$/),
 }).required()
 
+const schema = Joi.object({
+  message: Joi.string()
+    .regex(/^\d{1,3}%|unknown$/)
+    .required(),
+})
+
+const svgValueMatcher = />(\d{1,3}%|unknown)<\/text><\/g>/
+
+const badgeTokenPattern = /^\w{10}$/
+
 const documentation = `
   <p>
     You may specify a Codecov token to get coverage for a private repository.
   </p>
   <p>
-    See the <a href="https://docs.codecov.io/reference#authorization">Codecov Docs</a>
-    for more information about creating a token.
+    You can find the token for your badge in this url: <code>https://codecov.io/{vcsName}/{user}/{repo}/settings/badge</code>.
+  </p>
+  <p>
+    You can also use Codecov's API token (doesn't support <code>flag</code>). See the <a href="https://docs.codecov.io/reference#authorization">docs</a> for guidance to create API token.
```

Comment:
From what I'm seeing, one benefit of the Codecov badge token is that it is strictly read-only, whereas the API token (currently used for private repos) cannot be scoped within codecov; authenticating with the API token provides authorization to everything the token-owner has access to.

As I mentioned above, we do prefer API over SVG scraping, but in this instance badge scraping approach would be preferable in every case (except any existing private, flagless badges) because it allows users to use a badge-read-only token.


As such, let's not even mention the API auth token in the docs. We'll maintain support to avoid breaking existing badges, but lets steer folks to use the badge tokens.
```suggestion
    You can find the token under the badge section of your project settings page, in this url: <code>https://codecov.io/{vcsName}/{user}/{repo}/settings/badge</code>.
```